api/src/main/java/org/openmrs/module/fhir2/api/FhirLocationService.java

```diff
@@ -28,4 +30,6 @@
 	Collection<Location> findLocationsByPostalCode(@NotNull String postalCode);
 	
 	Collection<Location> findLocationsByState(@NotNull String state);
+	
+	Collection<Location> findLocationsByTags(@NotNull List<TokenParam> tags);
 }
```

Comment:
Instead of **`List<TokenParam> tags`** use **`TokenAndListParam tags`**

Reply:
Also, @NotNull doesn't apply to the list. You should be checking if it's empty

Reply:
I'd rather avoid using `TokenAndListParam` right now because things get a bit dicey (the `...AndListParam` contains `...OrListParam` entries which contain actual `TokenParam`s). I'm working on setting up a pattern to work with these types of things.

Reply:
This makes sense, I'm looking forward to improve this once you've finish setting up the enhanced search.